content/docs/1.5.4/high-availability/recover-volume.md

```diff
@@ -1,13 +1,33 @@
-Longhorn automatically deletes the workload pod if the pod is managed by a controller (e.g. deployment, statefulset, daemonset, etc...).
-By deleting the pod, its controller restarts the pod and Kubernetes handles volume reattachment and remount.
+## Automatically deletes the workload pod
 
-If you don't want Longhorn to automatically delete the workload pod, you can set it in [the setting `Automatically Delete Workload Pod when The Volume Is Detached Unexpectedly`](../../references/settings#automatically-delete-workload-pod-when-the-volume-is-detached-unexpectedly) in Longhorn UI.
+When the following situations occur, Longhorn automatically attempts to delete workload pods that are managed by a controller (for example, Deployment, StatefulSet, or DaemonSet). After deletion, the controller restarts the workload pod and Kubernetes handles volume reattachment and remounting.
 
-For the pods that don't have a controller, Longhorn doesn't delete them because if Longhorn does, no one will restart them.
-To recover unexpectedly detached volumes, you would have to manually delete and recreate the pods that don't have a controller.
+1. A volume was unexpectedly detached, possibly because of a [Kubernetes upgrade](https://github.com/longhorn/longhorn/issues/703), [container runtime reboot](https://github.com/longhorn/longhorn/issues/686), network connectivity issue, or volume engine crash.
+2. A volume was automatically salvaged after all replicas became faulty, possibly because of a network connectivity issue. Longhorn attempts to identify the usable replicas and uses them for the volume.
+3. An error occurred on a Share Manager pod that uses an RWX volume.
+
+If you want to prevent Longhorn from automatically deleting workload pods, disable the setting [*Automatically Delete Workload Pod when The Volume Is Detached Unexpectedly*](../../references/settings#automatically-delete-workload-pod-when-the-volume-is-detached-unexpectedly) on the Longhorn UI.
+
+Longhorn does not delete pods without a controller because such pods cannot be restarted after deletion. To recover volumes that are unexpectedly detached, you must manually delete and restart the pods without a controller.
+
+## Automatically remount the volume
+
+The state of a volume can change to read-only when IO errors occur. IO errors can be caused by a variety of issues, including the following:
+- Network disconnection: Interrupted connection between the engine and replicas.
+- High disk latency: Significant delay in the transfer of data between a replica and the corresponding disk.
+
+Longhorn checks the volume global mount point state every 10 sec. When the filesystem of the volume changes to read-only, Longhorn automatically attempts to remount the global mount point on the host to change the state back to read-write. Upon successful remounting, the workload pods continue functioning without disruption.
+
+## Conclusion
```

Comment:
More suggestions and info can be provided in this section. 
IIUC, IO errors may lead to engine crash then can be handled by `Automatically deletes the workload pod`. In this case, does users need to enable the setting?

Reply:
`Automatically remount the volume` this feature doesn't have setting to enable/disable it.
It just blindly tries to remount the `globalmount` when detecting `ro`

For engine crash case,
1. Longhorn will detach and reattach the the volume (no setting, default behavior)
2. The volume will become healthy again
3. However, the K8s won't perform remount in this case because the workload pod is still running 
4. User won't be able to perform IO and will get `read-only filesystem` error and the filesystem become `read-only`

In this case, the `Automatically remount the volume`  won't help, 
because when it tries to remount the filesystem it will get 
```
/host/var/lib/kubelet/plugins/kubernetes.io/csi/driver.longhorn.io/fe727eb53a0f2a72f000ffa85e652a92674a30d43befce23d0468712717b0de1/globalmount: 
cannot remount /dev/longhorn/pvc-e83db2c5-569e-46e0-add9-15db7debdab2 read-write, is write-protected.
```

Thus, for **engine crash case**, user still needs to enable `Automatically deletes the workload pod` to request k8s to remount the workload.

cc @derekbit

Reply:
I have added the case study to the  `Automatically remount the volume` section